lib/logstash/outputs/s3.rb

```diff
@@ -335,20 +383,29 @@ def shutdown_upload_workers
   end
 
   private
-  def handle_event(encoded_event)
+  def handle_event(encoded_event, event)
+    actualprefix = event.sprintf(@prefix)
+    if not @prefixes.to_a().include? actualprefix
```

Comment:
no need to convert the set into an array since **Set** class already support `include?`
```ruby
irb(main):002:0> require 'set'
=> true
irb(main):003:0> m = Set[1,2,3]
=> #<Set: {1, 2, 3}>
irb(main):004:0> m.include?(1)
=> true
irb(main):005:0>